### ISSUE TITLE
the dim of PluginTensorDesc is not dynamic when the trt file is generated using --minShape, optShape, maxShape 

### ISSUE THREAD

**Issue by SakuraRiven** [NVIDIA/TensorRT#2692]
## Description
Hi, we have a onnx model with dynamic axis and our own plugin. We hope that the model could handle data with arbitrary shapes. However, we found that the ```dims``` of ```inputDesc``` in our own plugin are always the maxShapes we pre-define. So, how could we get the correct shape and real tensor for each different inputs?

## Environment

**TensorRT Version**: 8.4.3.1
**NVIDIA GPU**: GA102 [GeForce RTX 3080 Lite Hash Rate]
**NVIDIA Driver Version**: 510.54
**CUDA Version**: 11.4
**CUDNN Version**: 8.0.5
**Operating System**: linux
**Python Version (if applicable)**: 3.9.12
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

## When we run the trt model, the dims (nums 1,2,3) in following plugin code are not dynamic, so as well as the f1,f2,f3

```
int OurPlugin::enqueue(
    const nvinfer1::PluginTensorDesc *inputDesc,
    const nvinfer1::PluginTensorDesc *outputDesc, const void *const *inputs,
    void *const *outputs, void *workSpace, cudaStream_t stream) noexcept {
  int num1 = inputDesc[0].dims.d[0];
  int num2 = inputDesc[0].dims.d[1];
  int num3 = outputDesc[0].dims.d[0];

  const void *f1 = inputs[0];
  const void *f2 = inputs[1];
  const void *f3 = inputs[2];

...
}
```

BTW, the inputs of plugin are 2D tensors and flatten

**Comment by zerollzeng** [comment]
I think you can refer to [configurePlugin()](https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_plugin_v2_dynamic_ext.html#acac4692dd8653d1fd428d282697b50ee)

**Comment by SakuraRiven** [comment]
> I think you can refer to [configurePlugin()](https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_plugin_v2_dynamic_ext.html#acac4692dd8653d1fd428d282697b50ee)

@zerollzeng  Thanks for your help. We try this funciton in our plugin but found that the nums1,2,3 are still the fixed max_value when we generate trt.

```
void OurPlugin::configurePlugin(
    const nvinfer1::DynamicPluginTensorDesc *inputs, int nbInputs,
    const nvinfer1::DynamicPluginTensorDesc *outputs, int nbOutputs) noexcept {

    num1 = inputs[0].desc.dims.d[0];
    num2 = inputs[0].desc.dims.d[1];
    num3 = inputs[1].desc.dims.d[0];
    printf("configurePlugin %d, %d, %d\n", num1, num2, num3);
}
```
The log is 
```
configurePlugin 150000, 32, 150000
configurePlugin 150000, 64, 150000
```
However, the real inputs are
```
configurePlugin 78383, 32, 12061
configurePlugin 78383, 64, 12061
```

@zerollzeng  Are there any problems in above code? BTW the inputs are not 4D tensor and the trt generation is with
```
--minShapes=input0:1x10,input1:1,input2:1 --optShapes=input0:78000x10,input1:40000,input2:78000 --maxShapes=input0:150000x10,input1:150000,input2:150000
```

**Comment by zerollzeng** [comment]
I feel like something is wrong with your plugin codes, can you try https://github.com/NVIDIA-AI-IOT/tensorrt_plugin_generator? use this tool to generate an empty plugin(no real enqueue) quickly and test. I think you should be able to find out the reason by comparing the generated codes and yours.

**Comment by SakuraRiven** [comment]
@zerollzeng Hi, we generate the official code but do not found obvious differences. We upload the onnx and hpp cpp files in [repo](https://github.com/SakuraRiven/Debug). Could you kindly help us finding if there are some problems? Thanks : )

**Comment by zerollzeng** [comment]
I did some experiments with the code you provided, I use TPG to generate plugin codes since your code has other dependencies which cannot be compiled directly. Here is the code: 
[ScattermaxpluginIPluginV2DynamicExt.tar.gz](https://github.com/NVIDIA/TensorRT/files/10785587/ScattermaxpluginIPluginV2DynamicExt.tar.gz)

I test it with trtexec, during inference, it's use optShapes:
```
[02/20/2023-15:04:14] [I] Starting inference
 ----> debug <---- call [ScattermaxpluginIPluginV2DynamicExt.cpp:256][configurePlugin]
 ----> debug <---- call [ScattermaxpluginIPluginV2DynamicExt.cpp:256][configurePlugin]
 ----> debug <---- call [ScattermaxpluginIPluginV2DynamicExt.cpp:274][enqueue]
78000 32 40000
 ----> debug <---- call [ScattermaxpluginIPluginV2DynamicExt.cpp:274][enqueue]
78000 64 40000
[02/20/2023-15:04:14] [I] Warmup completed 0 queries over 0 ms
[02/20/2023-15:04:14] [I] Timing trace has 1 queries over 0.0025567 s
[02/20/2023-15:04:14] [I]
[02/20/2023-15:04:14] [I] === Trace details ===
[02/20/2023-15:04:14] [I] Trace averages of 10 runs:
[02/20/2023-15:04:14] [I]
[02/20/2023-15:04:14] [I] === Performance summary ===
[02/20/2023-15:04:14] [I] Throughput: 391.129 qps
[02/20/2023-15:04:14] [I] Latency: min = 2.55254 ms, max = 2.55254 ms, mean = 2.55254 ms, median = 2.55254 ms, percentile(99%) = 2.55254 ms
[02/20/2023-15:04:14] [I] Enqueue Time: min = 0.550368 ms, max = 0.550368 ms, mean = 0.550368 ms, median = 0.550368 ms, percentile(99%) = 0.550368 ms
[02/20/2023-15:04:14] [I] H2D Latency: min = 0.147232 ms, max = 0.147232 ms, mean = 0.147232 ms, median = 0.147232 ms, percentile(99%) = 0.147232 ms
[02/20/2023-15:04:14] [I] GPU Compute Time: min = 2.11104 ms, max = 2.11104 ms, mean = 2.11104 ms, median = 2.11104 ms, percentile(99%) = 2.11104 ms
[02/20/2023-15:04:14] [I] D2H Latency: min = 0.294272 ms, max = 0.294272 ms, mean = 0.294272 ms, median = 0.294272 ms, percentile(99%) = 0.294272 ms
[02/20/2023-15:04:14] [I] Total Host Walltime: 0.0025567 s
[02/20/2023-15:04:14] [I] Total GPU Compute Time: 0.00211104 s
[02/20/2023-15:04:14] [I] Explanations of the performance metrics are printed in the verbose logs.
[02/20/2023-15:04:14] [I]
&&&& PASSED TensorRT.trtexec [TensorRT v8412] # /usr/src/tensorrt/bin/trtexec --onnx=model.onnx --plugins=ScattermaxpluginIPluginV2DynamicExt/libScattermaxpluginIPluginV2DynamicExt.so --minShapes=features:1x10,unq:1,unq_inv:1 --optShapes=features:78000x10,unq:40000,unq_inv:78000 --maxShapes=features:150000x10,unq:150000,unq_inv:150000 --duration=0 --iterations=1 --warmUp=0
```

Are you missing call the setInputShape (char const *tensorName, Dims const &dims) before you do the inference? please refer to https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_execution_context.html#a4f36f87c2ee13eef2b1fbb61397cb8bf

**Comment by SakuraRiven** [comment]
@zerollzeng Thanks for your suggestions. This function does deliver the correct shapes of real inputs. Solved : )

**Comment by zerollzeng** [comment]
You are welcome, can we close this issue? Thanks!
